### DISCUSSION TITLE
Ollama Offline Connection Failed Using Open-webui (even though Ollama itself works in Command Prompt)

### DISCUSSION THREAD

**Discussion by john911f** [open-webui/open-webui#5100] [General]
In an offline environment (ethernet cable unplugged):
- Ollama is running on http://localhost:11434/
- Ollama is running inside Cmd Prompt
- Ollama is NOT running in open-webui (specifically, llama models are NOT available)

In an online environment (ethernet cable plugged):
- Ollama is running in open-webui (specifically, llama models ARE available)

I am running Open-Webui manually in a Python environment, not through Docker.

Please help. Thanks a million!

### Discussed in https://github.com/open-webui/open-webui/discussions/1901

<div type='discussions-op-text'>

<sup>Originally posted by **start-life** May  2, 2024</sup>
Ollama offline connection failed
When it's online it connects

INFO:     Started server process [8800]
INFO:     Waiting for application startup.
INFO:apps.litellm.main:start_litellm_background
INFO:apps.litellm.main:run_background_process
INFO:apps.litellm.main:Executing command: ['litellm', '--port', '14365', '--host', '127.0.0.1', '--telemetry', 'False', '--config', 'C:\\zxcv\\open-webui\\backend\\data/litellm/config.yaml']
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
INFO:apps.litellm.main:Subprocess started successfully.
INFO:     127.0.0.1:51747 - "GET /api/config HTTP/1.1" 200 OK
INFO:     127.0.0.1:51748 - "GET /manifest.json HTTP/1.1" 200 OK
INFO:     127.0.0.1:51748 - "GET /api/v1/auths/ HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:     127.0.0.1:51748 - "GET /ollama/api/tags HTTP/1.1" 200 OK
INFO:apps.openai.main:get_all_models()
INFO:apps.openai.main:get_all_models()
INFO:     127.0.0.1:51748 - "GET /openai/api/models HTTP/1.1" 200 OK
INFO:     127.0.0.1:51748 - "GET /litellm/api/v1/models HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/v1/modelfiles/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/v1/prompts/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/v1/documents/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/v1/chats/tags/all HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:     127.0.0.1:51754 - "GET /ollama/api/tags HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/changelog HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
INFO:apps.openai.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:apps.openai.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:     127.0.0.1:51754 - "GET /openai/api/models HTTP/1.1" 200 OK
INFO:     127.0.0.1:51759 - "GET /ollama/api/version HTTP/1.1" 500 Internal Server Error
INFO:     127.0.0.1:51754 - "GET /static/favicon.png HTTP/1.1" 200 OK
INFO:     127.0.0.1:51759 - "GET /litellm/api/v1/models HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:     127.0.0.1:51761 - "GET /ollama/urls HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:     127.0.0.1:51761 - "GET /ollama/api/version HTTP/1.1" 500 Internal Server Error
INFO:     127.0.0.1:51761 - "GET /litellm/api/model/info HTTP/1.1" 200 OK</div>

**Comment by fullstackwebdev** [comment]
this is actually a bug it is trying to connect via SSL to http://

it should only connect via SSL if it is https <- NOTICE THE S

**Comment by natlamir** [comment]
Try this, I posted it in the other thread, it worked for me: https://github.com/open-webui/open-webui/discussions/1901#discussioncomment-10769700

**Reply by Solenhal** [reply]
Worked for me too.
